bin/watch-packages-and-rebuild.js

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+const fs = require( 'fs' );
+const path = require( 'path' );
+const { spawnSync } = require( 'child_process' );
+const chokidar = require( 'chokidar' );
+
+const packagesDirectoryPath = path.join( '.', 'packages' );
+const watcher = chokidar.watch( packagesDirectoryPath, {
+	ignored: /^\./,
+	persistent: true,
+	ignoreInitial: true,
+} );
+watcher.on( 'change', handleChange ).on( 'add', handleChange ).on( 'unlink', handleChange );
+
+let rebuildQueue = [];
+
+setTimeout( processRebuildQueue, 100 );
+
+function processRebuildQueue() {
+	rebuildQueue.map( rebuildPackage );
+	rebuildQueue = [];
+	setTimeout( processRebuildQueue, 100 );
+}
+
+function handleChange( filePath ) {
+	if ( isPathIgnored( filePath ) ) {
+		return;
+	}
+	console.log( 'heard change in', filePath );
+	const packageName = getPackageNameFromFilePath( filePath );
+	if ( packageName && ! rebuildQueue.includes( packageName ) ) {
+		rebuildQueue.push( packageName );
+	}
+}
+
+function isPathIgnored( filePath ) {
+	const filePathPieces = filePath.split( path.sep );
+	if ( filePathPieces.length < 2 ) {
+		return true;
+	}
+	const ignoredPaths = [ 'dist', 'test', 'tests' ];
+	if ( ignoredPaths.find( ( ignored ) => filePathPieces.includes( ignored ) ) ) {
+		return true;
+	}
+	const ignoredExtensions = [ '.css', '.md', '.d.ts' ];
+	if ( ignoredExtensions.find( ( ignored ) => filePath.endsWith( ignored ) ) ) {
+		return true;
+	}
+	return false;
+}
+
+function getPackageNameFromFilePath( filePath ) {
+	const relativeFilePath = path.relative( '.', filePath );
+	const relativeFilePathPieces = relativeFilePath.split( path.sep );
+	if ( relativeFilePathPieces.length < 2 ) {
+		console.error( 'failed to rebuild package: file is outside packages directory', filePath );
+		return null;
+	}
+	if ( relativeFilePathPieces[ 0 ] !== 'packages' ) {
+		console.error( 'failed to rebuild package: file is outside packages directory', filePath );
+		return null;
+	}
+
+	const packageJSONFile = path.join(
```

Comment:
If you agree with @scinos's idea of using the package dir, then you can return `relativeFilePathPieces[ 1 ]` here and be happy :D